src/datta_lab_to_nwb/markowitz_gillis_nature_2023/fiberphotometryinterface.py

```diff
@@ -0,0 +1,198 @@
+"""Primary class for converting fiber photometry data (dLight fluorescence)."""
+# Standard Scientific Python
+import pandas as pd
+
+# NWB Ecosystem
+from pynwb.file import NWBFile
+from pynwb.core import DynamicTableRegion
+from pynwb.ophys import RoiResponseSeries
+from ndx_photometry import (
+    FibersTable,
+    PhotodetectorsTable,
+    ExcitationSourcesTable,
+    DeconvolvedRoiResponseSeries,
+    MultiCommandedVoltage,
+    FiberPhotometry,
+    FluorophoresTable,
+)
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.utils import load_dict_from_file
+from neuroconv.tools import nwb_helpers
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+
+# Local
+
+
+class FiberPhotometryInterface(BaseDataInterface):
+    """Fiber Photometry  interface for markowitz_gillis_nature_2023 conversion"""
+
+    def __init__(self, file_path: str, session_uuid: str, metadata_path: str):
+        # This should load the data lazily and prepare variables you need
+        columns = (
+            "uuid",
+            "signal_dff",
+            "reference_dff",
+            "uv_reference_fit",
+            "reference_dff_fit",
+            "timestamp",
+        )
+        super().__init__(
+            file_path=file_path,
+            session_uuid=session_uuid,
+            columns=columns,
+            metadata_path=metadata_path,
+        )
+
+    def get_metadata(self) -> dict:
+        metadata = super().get_metadata()
+        session_metadata = load_dict_from_file(self.source_data["metadata_path"])
+        session_metadata = session_metadata[self.source_data["session_uuid"]]
+        metadata["NWBFile"]["session_description"] = session_metadata["session_description"]
+        metadata["NWBFile"]["session_start_time"] = session_metadata["session_start_time"]
+        metadata["NWBFile"]["identifier"] = self.source_data["session_uuid"]
+        metadata["NWBFile"]["session_id"] = self.source_data["session_uuid"]
+        metadata["Subject"] = {}
+        metadata["Subject"]["subject_id"] = session_metadata["subject_id"]
+        metadata["FiberPhotometry"] = {}
```

Comment:
Fun fact: if you're using the current dev branch of NeuroConv (not previous release) we use a `DeepDict` structure that let's you skip lines like this and go straight to nested `dict[key1][key2][...]=value` setting